python/paddle/fluid/tests/unittests/dygraph_to_static/test_lac.py

```diff
@@ -0,0 +1,570 @@
+# Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+import math
+import time
+import numpy as np
+import unittest
+
+import paddle.fluid as fluid
+from paddle.fluid.dygraph import to_variable
+from paddle.fluid.dygraph import Embedding, Linear, GRUUnit
+from paddle.fluid.dygraph import declarative, ProgramTranslator
+
+SEED = 2020
+program_translator = ProgramTranslator()
+
+
+class DynamicGRU(fluid.dygraph.Layer):
+    def __init__(self,
+                 size,
+                 h_0=None,
+                 param_attr=None,
+                 bias_attr=None,
+                 is_reverse=False,
+                 gate_activation='sigmoid',
+                 candidate_activation='tanh',
+                 origin_mode=False,
+                 init_size=None):
+        super(DynamicGRU, self).__init__()
+
+        self.gru_unit = GRUUnit(
+            size * 3,
+            param_attr=param_attr,
+            bias_attr=bias_attr,
+            activation=candidate_activation,
+            gate_activation=gate_activation,
+            origin_mode=origin_mode)
+
+        self.size = size
+        self.h_0 = h_0
+        self.is_reverse = is_reverse
+
+    def forward(self, inputs):
+        inputs = to_variable(inputs)
+        hidden = to_variable(self.h_0)
+        hidden.stop_gradient = True
+        res = []
+
+        for i in range(inputs.shape[1]):
+            if self.is_reverse:
+                j = fluid.layers.shape(inputs)[1] - 1 - i
+            else:
+                # TODO(Aurelius84): In while block, if the var created in parent block
+                # participates in the calculation of gradient, the result of gradient
+                # is incorrect because each step scope always returns the same value
+                # generated by last step. Here we add 0 to create `j` in while block to
+                # avoid this bug, and working on fixing it in next PR.
+                j = i + 0
+
+            if i == 0:
+                h_0 = to_variable(self.h_0)
+            else:
+                h_0 = hidden
+
+            input_ = fluid.layers.slice(
+                inputs, axes=[1], starts=[j], ends=[j + 1])
```

Comment:
Better add a code comment to explain why not use `inputs[:,i:i+1,:]`

Reply:
Thanks, done.